docs/appendices/release-notes/unreleased.rst

```diff
@@ -111,4 +111,9 @@ Fixes
 .. stable branch. You can add a version label (`v/X.Y`) to the pull request for
 .. an automated mergify backport.
 
-None
+- Fixed an issue that caused ``UNION`` and ``GROUP BY`` statements containing
+  constant ``NULL`` as union field or grouping key throw an error. Examples:
```

Comment:
not sure all examples has to be listed - especially 2 group by-s : I added third as it was failing even when 2 first cases were passing, so treated it differently but for user it's just group by - maybe just keep union and last simpler group by?